--- glusterfs-3.7.9/configure.ac.orig	2016-03-31 21:42:00.820154766 +0200
+++ glusterfs-3.7.9/configure.ac	2016-04-01 17:09:46.650547715 +0200
@@ -1203,7 +1203,7 @@
 esac
 
 if test "x$BUILD_GLUPY" = "xyes"; then
-   BUILD_PYTHON_SITE_PACKAGES=`$PYTHON -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib())'`
+   BUILD_PYTHON_SITE_PACKAGES=`$PYTHON -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib(prefix="'$prefix'"))'`
    BUILD_PYTHON_INC=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`
    BUILD_PYTHON_LIB=python$PYTHON_VERSION
    GLUPY_SUBDIR=glupy
