--- glusterfs-3.4.0/glusterfs-api.pc.in.orig	2013-07-12 19:43:26.000000000 +0200
+++ glusterfs-3.4.0/glusterfs-api.pc.in	2013-07-23 21:33:24.416239174 +0200
@@ -8,5 +8,5 @@
 Description: GlusterFS API
 /* This is the API version, NOT package version */
 Version: 4
-Libs: -Wl,-R${libdir} -L${libdir} -lgfapi -lglusterfs -lgfrpc -lgfxdr
+Libs: -L${libdir} -lgfapi -lglusterfs -lgfrpc -lgfxdr
 Cflags: -I${includedir}/glusterfs -D_FILE_OFFSET_BITS=64 -D__USE_FILE_OFFSET64
