diff -ur glusterfs-2.0.1-o/transport/ib-verbs/src/Makefile.am glusterfs-2.0.1/transport/ib-verbs/src/Makefile.am
--- glusterfs-2.0.1-o/transport/ib-verbs/src/Makefile.am	2009-06-03 21:43:45.000000000 -0600
+++ glusterfs-2.0.1/transport/ib-verbs/src/Makefile.am	2009-06-03 22:33:02.000000000 -0600
@@ -1,7 +1,7 @@
 noinst_HEADERS = ib-verbs.h name.h
 
 transport_LTLIBRARIES = ib-verbs.la
-transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/
+transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport
 
 ib_verbs_la_LDFLAGS = -module -avoid-version
 
diff -ur glusterfs-2.0.1-o/transport/socket/src/Makefile.am glusterfs-2.0.1/transport/socket/src/Makefile.am
--- glusterfs-2.0.1-o/transport/socket/src/Makefile.am	2009-06-03 21:43:45.000000000 -0600
+++ glusterfs-2.0.1/transport/socket/src/Makefile.am	2009-06-03 22:33:04.000000000 -0600
@@ -1,7 +1,7 @@
 noinst_HEADERS = socket.h name.h
 
 transport_LTLIBRARIES = socket.la
-transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/
+transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport
 
 socket_la_LDFLAGS = -module -avoid-version
 
