--- glusterfs-3.10.1/configure.ac.orig	2017-04-16 17:57:27.942310649 +0200
+++ glusterfs-3.10.1/configure.ac	2017-04-16 18:03:50.218972950 +0200
@@ -521,7 +521,7 @@
    PYTHONDEV_CPPFLAGS=`echo ${PYTHONDEV_CPPFLAGS} | sed -e 's/-Wp,-D_FORTIFY_SOURCE=[[0-9]]//g'`
 
    dnl Find python libs at user configured libdir and also "lib" under prefix
-   PYTHONDEV_LDFLAGS="${PYTHON_LIBS} -L`${PYTHON}-config --prefix`/lib -L`${PYTHON}-config --prefix`/$libdir `${PYTHON}-config --ldflags`"
+   PYTHONDEV_LDFLAGS="${PYTHON_LIBS} `${PYTHON}-config --ldflags`"
 
    BUILD_PYTHON_SITE_PACKAGES=${pythondir}
    BUILD_PYTHON_INC=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_inc())" 2>/dev/null`
