diff -dur glusterfs-3.8.11.orig/configure.ac glusterfs-3.8.11/configure.ac
--- glusterfs-3.8.11.orig/configure.ac	2017-04-21 14:47:20.000000000 +0200
+++ glusterfs-3.8.11/configure.ac	2017-04-21 14:48:27.000000000 +0200
@@ -1150,7 +1150,7 @@
     PYTHONDEV_CPPFLAGS=`echo ${PYTHONDEV_CPPFLAGS} | sed -e 's/-Wp,-D_FORTIFY_SOURCE=[[0-9]]//g'`
 
     dnl Find python libs at user configured libdir and also "lib" under prefix
-    PYTHONDEV_LDFLAGS="${PYTHON_LIBS} -L`${PYTHON}-config --prefix`/lib -L`${PYTHON}-config --prefix`/$libdir `${PYTHON}-config --ldflags`"
+    PYTHONDEV_LDFLAGS="${PYTHON_LIBS} `${PYTHON}-config --ldflags`"
 
     BUILD_PYTHON_SITE_PACKAGES=${pythondir}
     BUILD_PYTHON_INC=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_inc())" 2>/dev/null`
