--- glusterfs-1.3.10/xlators/cluster/unify/src/Makefile.am-fix_link	2008-08-13 20:03:32.000000000 +0200
+++ glusterfs-1.3.10/xlators/cluster/unify/src/Makefile.am	2008-08-13 20:05:00.000000000 +0200
@@ -15,12 +15,15 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = unify.so
+xlator_LTLIBRARIES = unify.la
+
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/cluster
 
-unify_so_SOURCES = unify.c unify-self-heal.c
+unify_la_SOURCES = unify.c unify-self-heal.c
 noinst_HEADERS = unify.h
 
+unify_la_LDFLAGS = -module -avoid-version
+
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
 
--- glusterfs-1.3.10/xlators/cluster/afr/src/Makefile.am-fix_link	2008-08-13 20:08:44.000000000 +0200
+++ glusterfs-1.3.10/xlators/cluster/afr/src/Makefile.am	2008-08-13 20:09:28.000000000 +0200
@@ -14,12 +14,14 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = afr.so
+xlator_LTLIBRARIES = afr.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/cluster
 
-afr_so_SOURCES = afr.c afr_self_heal.c
+afr_la_SOURCES = afr.c afr_self_heal.c
 noinst_HEADERS = afr.h
 
+afr_la_LDFLAGS = -module -avoid-version
+
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
 	    -I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
 
--- glusterfs-1.3.10/xlators/cluster/stripe/src/Makefile.am-fix_link	2008-08-13 20:06:59.000000000 +0200
+++ glusterfs-1.3.10/xlators/cluster/stripe/src/Makefile.am	2008-08-13 20:07:55.000000000 +0200
@@ -15,10 +15,13 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = stripe.so
+xlator_LTLIBRARIES  = stripe.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/cluster
 
-stripe_so_SOURCES = stripe.c
+stripe_la_SOURCES = stripe.c
+
+stripe_la_LDFLAGS = -module -avoid-version
+
 #noinst_HEADERS = unify.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/xlators/debug/trace/src/Makefile.am-fix_link	2008-08-14 10:11:45.000000000 +0200
+++ glusterfs-1.3.10/xlators/debug/trace/src/Makefile.am	2008-08-14 10:12:33.000000000 +0200
@@ -15,10 +15,11 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = trace.so
+xlator_LTLIBRARIES = trace.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/debug
 
-trace_so_SOURCES = trace.c
+trace_la_SOURCES = trace.c
+trace_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/xlators/features/trash/src/Makefile.am-fix_link	2008-08-14 10:24:13.000000000 +0200
+++ glusterfs-1.3.10/xlators/features/trash/src/Makefile.am	2008-08-14 10:24:59.000000000 +0200
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = trash.so
+xlator_LTLIBRARIES = trash.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/features
 
-trash_so_SOURCES = trash.c
+trash_la_SOURCES = trash.c
+trash_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/xlators/features/filter/src/Makefile.am-fix_link	2008-08-14 10:13:46.000000000 +0200
+++ glusterfs-1.3.10/xlators/features/filter/src/Makefile.am	2008-08-14 10:14:21.000000000 +0200
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = filter.so
+xlator_LTLIBRARIES = filter.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/features
 
-filter_so_SOURCES = filter.c
+filter_la_SOURCES = filter.c
+filter_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = filter.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/xlators/features/fixed-id/src/Makefile.am-fix_link	2008-08-14 10:15:04.000000000 +0200
+++ glusterfs-1.3.10/xlators/features/fixed-id/src/Makefile.am	2008-08-14 10:21:35.000000000 +0200
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = fixed-id.so
+xlator_LTLIBRARIES = fixed-id.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/features
 
-fixed_id_so_SOURCES = fixed-id.c
+fixed_id_la_SOURCES = fixed-id.c
+fixed_id_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/xlators/features/posix-locks/src/Makefile.am-fix_link	2008-08-14 10:22:30.000000000 +0200
+++ glusterfs-1.3.10/xlators/features/posix-locks/src/Makefile.am	2008-08-14 10:23:02.000000000 +0200
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = posix-locks.so
+xlator_LTLIBRARIES = posix-locks.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/features
 
-posix_locks_so_SOURCES = posix-locks.c
+posix_locks_la_SOURCES = posix-locks.c
+posix_locks_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = posix-locks.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
--- glusterfs-1.3.10/xlators/mount/fuse/src/Makefile.am-fix_link	2008-08-14 10:27:51.000000000 +0200
+++ glusterfs-1.3.10/xlators/mount/fuse/src/Makefile.am	2008-08-14 10:36:43.000000000 +0200
@@ -17,15 +17,18 @@
 
 noinst_HEADERS = fuse-extra.h 
 
-xlator_PROGRAMS = fuse.so
+xlator_LTLIBRARIES = fuse.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/mount
-fuse_so_SOURCES = fuse-bridge.c fuse-extra.c
-fuse_so_CFLAGS = -DFUSE_USE_VERSION=26 -fPIC -Wall -pthread \
-	$(GF_DARWIN_BUNDLE_CFLAGS) -I$(top_srcdir)/libglusterfs/src \
-	-D$(GF_HOST_OS) -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE  \
-	-DDATADIR=\"$(localstatedir)\" -DCONFDIR=\"$(sysconfdir)/glusterfs\"
-fuse_so_LDFLAGS = -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
-fuse_so_LDADD = -lfuse $(GF_DARWIN_FUSE_LDADD)
+fuse_la_SOURCES = fuse-bridge.c fuse-extra.c
+
+#fuse_la_CFLAGS = -DFUSE_USE_VERSION=26 -fPIC -Wall -pthread \
+#	$(GF_DARWIN_BUNDLE_CFLAGS) -I$(top_srcdir)/libglusterfs/src \
+#	-D$(GF_HOST_OS) -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE  \
+#	-DDATADIR=\"$(localstatedir)\" -DCONFDIR=\"$(sysconfdir)/glusterfs\"
+
+fuse_la_LDFLAGS =  -module -avoid-version
+
+# fuse_la_LIBADD = -lfuse $(GF_DARWIN_FUSE_LDADD)
 
 
 CLEANFILES = *~
--- glusterfs-1.3.10/xlators/protocol/client/src/Makefile.am-fix_link	2008-08-13 20:17:53.000000000 +0200
+++ glusterfs-1.3.10/xlators/protocol/client/src/Makefile.am	2008-08-13 20:18:23.000000000 +0200
@@ -15,10 +15,12 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = client.so
+xlator_LTLIBRARIES = client.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/protocol
 
-client_so_SOURCES = client-protocol.c
+client_la_SOURCES = client-protocol.c
+client_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = client-protocol.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
--- glusterfs-1.3.10/xlators/protocol/server/src/Makefile.am-fix_link	2008-08-13 20:14:16.000000000 +0200
+++ glusterfs-1.3.10/xlators/protocol/server/src/Makefile.am	2008-08-13 20:17:06.000000000 +0200
@@ -15,10 +15,12 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = server.so
+xlator_LTLIBRARIES = server.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/protocol
 
-server_so_SOURCES = server-protocol.c
+server_la_SOURCES = server-protocol.c
+server_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = server-protocol.h 
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
--- glusterfs-1.3.10/xlators/storage/posix/src/Makefile.am-fix_link	2008-08-13 20:10:23.000000000 +0200
+++ glusterfs-1.3.10/xlators/storage/posix/src/Makefile.am	2008-08-13 20:10:57.000000000 +0200
@@ -15,12 +15,14 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
-xlator_PROGRAMS = posix.so
+xlator_LTLIBRARIES = posix.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/storage
 
-posix_so_SOURCES = posix.c 
+posix_la_SOURCES = posix.c 
 noinst_HEADERS = posix.h 
 
+posix_la_LDFLAGS = -module -avoid-version
+
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D$(GF_HOST_OS) -Wall \
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles \
 	$(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/xlators/encryption/rot-13/src/Makefile.am-fix_link	2008-08-14 10:26:13.000000000 +0200
+++ glusterfs-1.3.10/xlators/encryption/rot-13/src/Makefile.am	2008-08-14 10:26:46.000000000 +0200
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = rot-13.so
+xlator_LTLIBRARIES = rot-13.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/encryption
 
-rot_13_so_SOURCES = rot-13.c
+rot_13_la_SOURCES = rot-13.c
+rot_13_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = rot-13.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
--- glusterfs-1.3.10/xlators/performance/write-behind/src/Makefile.am-fix_link	2008-08-13 20:19:41.000000000 +0200
+++ glusterfs-1.3.10/xlators/performance/write-behind/src/Makefile.am	2008-08-13 20:20:12.000000000 +0200
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = write-behind.so
+xlator_LTLIBRARIES = write-behind.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/performance
 
-write_behind_so_SOURCES = write-behind.c
+write_behind_la_SOURCES = write-behind.c
+write_behind_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/xlators/performance/booster/src/Makefile.am-fix_link	2008-08-13 20:25:12.000000000 +0200
+++ glusterfs-1.3.10/xlators/performance/booster/src/Makefile.am	2008-08-14 10:05:45.000000000 +0200
@@ -14,15 +14,16 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = booster.so
+xlator_LTLIBRARIES = booster.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/performance
 
-booster_so_SOURCES = booster.c
+booster_la_SOURCES = booster.c
 
-booster_so_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall \
+booster_la_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall \
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles \
 	-D$(GF_HOST_OS) $(GF_DARWIN_BUNDLE_CFLAGS)
-booster_so_LDFLAGS = -shared $(GF_DARWIN_BUNDLE_CFLAGS)
+booster_la_LDFLAGS = -shared $(GF_DARWIN_BUNDLE_CFLAGS) \
+	-module -avoid-version
 
 noinst_HEADERS = booster.h
 
--- glusterfs-1.3.10/xlators/performance/io-cache/src/Makefile.am-fix_link	2008-08-13 20:23:53.000000000 +0200
+++ glusterfs-1.3.10/xlators/performance/io-cache/src/Makefile.am	2008-08-13 20:24:33.000000000 +0200
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = io-cache.so
+xlator_LTLIBRARIES = io-cache.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/performance
 
-io_cache_so_SOURCES = io-cache.c page.c ioc-inode.c
+io_cache_la_SOURCES = io-cache.c page.c ioc-inode.c
+io_cache_la_LDFLAGS = -module -avoid-version
 
 noinst_HEADERS = io-cache.h
 
--- glusterfs-1.3.10/xlators/performance/read-ahead/src/Makefile.am-fix_link	2008-08-13 20:20:54.000000000 +0200
+++ glusterfs-1.3.10/xlators/performance/read-ahead/src/Makefile.am	2008-08-13 20:22:02.000000000 +0200
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = read-ahead.so
+xlator_LTLIBRARIES = read-ahead.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/performance
 
-read_ahead_so_SOURCES = read-ahead.c page.c
+read_ahead_la_SOURCES = read-ahead.c page.c
+read_ahead_la_LDFLAGS = -module -avoid-version
 
 noinst_HEADERS = read-ahead.h
 
--- glusterfs-1.3.10/xlators/performance/io-threads/src/Makefile.am-fix_link	2008-08-13 20:22:43.000000000 +0200
+++ glusterfs-1.3.10/xlators/performance/io-threads/src/Makefile.am	2008-08-13 20:23:09.000000000 +0200
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-xlator_PROGRAMS = io-threads.so
+xlator_LTLIBRARIES = io-threads.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/performance
 
-io_threads_so_SOURCES = io-threads.c
+io_threads_la_SOURCES = io-threads.c
+io_threads_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = io-threads.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/auth/ip/src/Makefile.am-link_fix	2008-08-14 11:50:00.000000000 +0000
+++ glusterfs-1.3.10/auth/ip/src/Makefile.am	2008-08-14 11:50:54.000000000 +0000
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-auth_PROGRAMS = ip.so
+auth_LTLIBRARIES = ip.la
 authdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/auth
 
-ip_so_SOURCES = ip.c
+ip_la_SOURCES = ip.c
+ip_la_LDFLAGS = -module -avoid-version
+
 #noinst_HEADERS = alu.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/scheduler/switch/src/Makefile.am-link_fix	2008-08-14 11:38:54.000000000 +0000
+++ glusterfs-1.3.10/scheduler/switch/src/Makefile.am	2008-08-14 11:39:26.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-sched_PROGRAMS = switch.so
+sched_LTLIBRARIES = switch.la
 scheddir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/scheduler
 
-switch_so_SOURCES = switch.c
+switch_la_SOURCES = switch.c
+switch_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/scheduler/nufa/src/Makefile.am-link_fix	2008-08-14 11:36:34.000000000 +0000
+++ glusterfs-1.3.10/scheduler/nufa/src/Makefile.am	2008-08-14 11:36:51.000000000 +0000
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-sched_PROGRAMS = nufa.so
+sched_LTLIBRARIES = nufa.la
 scheddir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/scheduler
 
-nufa_so_SOURCES = nufa.c
+nufa_la_SOURCES = nufa.c
+nufa_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = nufa.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/scheduler/alu/src/Makefile.am-link_fix	2008-08-14 11:32:40.000000000 +0000
+++ glusterfs-1.3.10/scheduler/alu/src/Makefile.am	2008-08-14 11:34:35.000000000 +0000
@@ -14,11 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-sched_PROGRAMS = alu.so
+sched_LTLIBRARIES = alu.la
 scheddir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/scheduler
 
-alu_so_SOURCES = alu.c
+alu_la_SOURCES = alu.c
 noinst_HEADERS = alu.h
+alu_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/scheduler/rr/src/Makefile.am-link_fix	2008-08-14 11:37:30.000000000 +0000
+++ glusterfs-1.3.10/scheduler/rr/src/Makefile.am	2008-08-14 11:38:28.000000000 +0000
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-sched_PROGRAMS = rr.so
+sched_LTLIBRARIES = rr.la
 scheddir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/scheduler
 
-rr_so_SOURCES = rr.c
+rr_la_SOURCES = rr.c
+rr_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = rr.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS) \
--- glusterfs-1.3.10/scheduler/random/src/Makefile.am-link_fix	2008-08-14 11:35:51.000000000 +0000
+++ glusterfs-1.3.10/scheduler/random/src/Makefile.am	2008-08-14 11:36:11.000000000 +0000
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-sched_PROGRAMS = random.so
+sched_LTLIBRARIES = random.la
 scheddir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/scheduler
 
-random_so_SOURCES = random.c
+random_la_SOURCES = random.c
+random_la_LDFLAGS = -module -avoid-version
+
 noinst_HEADERS = random.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
--- glusterfs-1.3.10/xlators/mount/fuse/src/Makefile.am-link_fix	2008-08-14 11:25:04.000000000 +0000
+++ glusterfs-1.3.10/xlators/mount/fuse/src/Makefile.am	2008-08-14 11:31:24.000000000 +0000
@@ -21,14 +21,15 @@ xlator_LTLIBRARIES = fuse.la
 xlatordir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/xlator/mount
 fuse_la_SOURCES = fuse-bridge.c fuse-extra.c
 
-#fuse_la_CFLAGS = -DFUSE_USE_VERSION=26 -fPIC -Wall -pthread \
-#	$(GF_DARWIN_BUNDLE_CFLAGS) -I$(top_srcdir)/libglusterfs/src \
-#	-D$(GF_HOST_OS) -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE  \
-#	-DDATADIR=\"$(localstatedir)\" -DCONFDIR=\"$(sysconfdir)/glusterfs\"
+fuse_la_CFLAGS = -DFUSE_USE_VERSION=26 -fPIC -Wall -pthread \
+	$(GF_DARWIN_BUNDLE_CFLAGS) -I$(top_srcdir)/libglusterfs/src \
+	-D$(GF_HOST_OS) -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE  \
+	-DDATADIR=\"$(localstatedir)\" -DCONFDIR=\"$(sysconfdir)/glusterfs\"
 
 fuse_la_LDFLAGS =  -module -avoid-version
 
 # fuse_la_LIBADD = -lfuse $(GF_DARWIN_FUSE_LDADD)
+fuse_la_LIBADD = -lfuse $(GF_DARWIN_FUSE_LDADD)
 
 
 CLEANFILES = *~
--- glusterfs-1.3.10/transport/tcp/client/src/Makefile.am-link_fix	2008-08-14 11:40:39.000000000 +0000
+++ glusterfs-1.3.10/transport/tcp/client/src/Makefile.am	2008-08-14 11:41:31.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = client.so
+transport_LTLIBRARIES = client.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/tcp/
 
-client_so_SOURCES = tcp-client.c $(top_srcdir)/transport/tcp/tcp.c
+client_la_SOURCES = tcp-client.c $(top_srcdir)/transport/tcp/tcp.c
+client_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/tcp/ -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/tcp/server/src/Makefile.am-link_fix	2008-08-14 11:42:13.000000000 +0000
+++ glusterfs-1.3.10/transport/tcp/server/src/Makefile.am	2008-08-14 11:42:36.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = server.so
+transport_LTLIBRARIES = server.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/tcp/
 
-server_so_SOURCES = tcp-server.c $(top_srcdir)/transport/tcp/tcp.c
+server_la_SOURCES = tcp-server.c $(top_srcdir)/transport/tcp/tcp.c
+server_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/tcp/ -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/ib-sdp/client/src/Makefile.am-link_fix	2008-08-14 11:43:08.000000000 +0000
+++ glusterfs-1.3.10/transport/ib-sdp/client/src/Makefile.am	2008-08-14 11:43:29.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = client.so
+transport_LTLIBRARIES = client.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/ib-sdp
 
-client_so_SOURCES = ib-sdp-client.c $(top_srcdir)/transport/ib-sdp/ib-sdp.c
+client_la_SOURCES = ib-sdp-client.c $(top_srcdir)/transport/ib-sdp/ib-sdp.c
+client_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/ib-sdp -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/ib-sdp/server/src/Makefile.am-link_fix	2008-08-14 11:43:52.000000000 +0000
+++ glusterfs-1.3.10/transport/ib-sdp/server/src/Makefile.am	2008-08-14 11:44:08.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = server.so
+transport_LTLIBRARIES = server.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/ib-sdp
 
-server_so_SOURCES = ib-sdp-server.c $(top_srcdir)/transport/ib-sdp/ib-sdp.c
+server_la_SOURCES = ib-sdp-server.c $(top_srcdir)/transport/ib-sdp/ib-sdp.c
+server_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/ib-sdp -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/unix/client/src/Makefile.am-link_fix	2008-08-14 11:44:57.000000000 +0000
+++ glusterfs-1.3.10/transport/unix/client/src/Makefile.am	2008-08-14 11:45:23.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = client.so
+transport_LTLIBRARIES = client.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/unix/
 
-client_so_SOURCES = unix-client.c $(top_srcdir)/transport/unix/unix.c
+client_la_SOURCES = unix-client.c $(top_srcdir)/transport/unix/unix.c
+client_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/unix/ -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/unix/server/src/Makefile.am-link_fix	2008-08-14 11:46:11.000000000 +0000
+++ glusterfs-1.3.10/transport/unix/server/src/Makefile.am	2008-08-14 11:46:27.000000000 +0000
@@ -14,10 +14,11 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = server.so
+transport_LTLIBRARIES = server.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/unix/
 
-server_so_SOURCES = unix-server.c $(top_srcdir)/transport/unix/unix.c
+server_la_SOURCES = unix-server.c $(top_srcdir)/transport/unix/unix.c
+server_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/unix/ -shared -nostartfiles $(GF_DARWIN_BUNDLE_CFLAGS)
--- glusterfs-1.3.10/transport/ib-verbs/client/src/Makefile.am-link_fix	2008-08-14 11:48:16.000000000 +0000
+++ glusterfs-1.3.10/transport/ib-verbs/client/src/Makefile.am	2008-08-14 11:48:40.000000000 +0000
@@ -14,11 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = client.so
+transport_LTLIBRARIES = client.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/ib-verbs
 
-client_so_SOURCES = ib-verbs-client.c $(top_srcdir)/transport/ib-verbs/ib-verbs.c
-client_so_LDADD = -libverbs
+client_la_SOURCES = ib-verbs-client.c $(top_srcdir)/transport/ib-verbs/ib-verbs.c
+client_la_LIBADD = -libverbs
+client_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/ib-verbs -shared -nostartfiles
--- glusterfs-1.3.10/transport/ib-verbs/server/src/Makefile.am-link_fix	2008-08-14 11:47:01.000000000 +0000
+++ glusterfs-1.3.10/transport/ib-verbs/server/src/Makefile.am	2008-08-14 11:47:35.000000000 +0000
@@ -14,11 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-transport_PROGRAMS = server.so
+transport_LTLIBRARIES = server.la
 transportdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/transport/ib-verbs
 
-server_so_SOURCES = ib-verbs-server.c $(top_srcdir)/transport/ib-verbs/ib-verbs.c
-server_so_LDADD = -libverbs
+server_la_SOURCES = ib-verbs-server.c $(top_srcdir)/transport/ib-verbs/ib-verbs.c
+server_la_LIBADD = -libverbs
+server_la_LDFLAGS = -module -avoid-version
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
 	-I$(top_srcdir)/libglusterfs/src -I$(top_srcdir)/transport/ib-verbs -shared -nostartfiles
--- glusterfs-1.3.10/auth/login/src/Makefile.am-link_fix	2008-08-14 12:01:52.000000000 +0000
+++ glusterfs-1.3.10/auth/login/src/Makefile.am	2008-08-14 12:02:38.000000000 +0000
@@ -14,10 +14,12 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-auth_PROGRAMS = login.so
+auth_LTLIBRARIES = login.la
 authdir = $(libdir)/glusterfs/$(PACKAGE_VERSION)/auth
 
-login_so_SOURCES = login.c
+login_la_SOURCES = login.c
+login_la_LDFLAGS = -module -avoid-version
+
 #noinst_HEADERS = alu.h
 
 AM_CFLAGS = -fPIC -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -Wall -D$(GF_HOST_OS)\
