diff -dur glusterfs-3.8.11.orig/extras/geo-rep/schedule_georep.py.in glusterfs-3.8.11/extras/geo-rep/schedule_georep.py.in
--- glusterfs-3.8.11.orig/extras/geo-rep/schedule_georep.py.in	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/extras/geo-rep/schedule_georep.py.in	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 """
 Schedule Geo-replication
 ------------------------
diff -dur glusterfs-3.8.11.orig/extras/snap_scheduler/gcron.py glusterfs-3.8.11/extras/snap_scheduler/gcron.py
--- glusterfs-3.8.11.orig/extras/snap_scheduler/gcron.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/extras/snap_scheduler/gcron.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com>
 # This file is part of GlusterFS.
diff -dur glusterfs-3.8.11.orig/extras/snap_scheduler/snap_scheduler.py glusterfs-3.8.11/extras/snap_scheduler/snap_scheduler.py
--- glusterfs-3.8.11.orig/extras/snap_scheduler/snap_scheduler.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/extras/snap_scheduler/snap_scheduler.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com>
 # This file is part of GlusterFS.
--- glusterfs-3.10.1/extras/stop-all-gluster-processes.sh.orig	2017-04-27 22:09:07.287953034 +0200
+++ glusterfs-3.10.1/extras/stop-all-gluster-processes.sh	2017-04-27 22:10:38.047952000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
 #
 # Kill all the processes/services except glusterd
 #
diff -dur glusterfs-3.8.11.orig/geo-replication/src/peer_mountbroker.in glusterfs-3.8.11/geo-replication/src/peer_mountbroker.in
--- glusterfs-3.8.11.orig/geo-replication/src/peer_mountbroker.in	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/geo-replication/src/peer_mountbroker.in	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 import os
 from argparse import ArgumentParser, RawDescriptionHelpFormatter
 import json
diff -dur glusterfs-3.8.11.orig/tools/gfind_missing_files/gfid_to_path.py glusterfs-3.8.11/tools/gfind_missing_files/gfid_to_path.py
--- glusterfs-3.8.11.orig/tools/gfind_missing_files/gfid_to_path.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/gfind_missing_files/gfid_to_path.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
 # This file is part of GlusterFS.
diff -dur glusterfs-3.8.11.orig/tools/glusterfind/glusterfind.in glusterfs-3.8.11/tools/glusterfind/glusterfind.in
--- glusterfs-3.8.11.orig/tools/glusterfind/glusterfind.in	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/glusterfind/glusterfind.in	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
 # This file is part of GlusterFS.
diff -dur glusterfs-3.8.11.orig/tools/glusterfind/src/brickfind.py glusterfs-3.8.11/tools/glusterfind/src/brickfind.py
--- glusterfs-3.8.11.orig/tools/glusterfind/src/brickfind.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/glusterfind/src/brickfind.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
diff -dur glusterfs-3.8.11.orig/tools/glusterfind/src/changelog.py glusterfs-3.8.11/tools/glusterfind/src/changelog.py
--- glusterfs-3.8.11.orig/tools/glusterfind/src/changelog.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/glusterfind/src/changelog.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
diff -dur glusterfs-3.8.11.orig/tools/glusterfind/src/main.py glusterfs-3.8.11/tools/glusterfind/src/main.py
--- glusterfs-3.8.11.orig/tools/glusterfind/src/main.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/glusterfind/src/main.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
diff -dur glusterfs-3.8.11.orig/tools/glusterfind/src/nodeagent.py glusterfs-3.8.11/tools/glusterfind/src/nodeagent.py
--- glusterfs-3.8.11.orig/tools/glusterfind/src/nodeagent.py	2017-04-11 10:33:35.000000000 +0200
+++ glusterfs-3.8.11/tools/glusterfind/src/nodeagent.py	2017-04-21 14:55:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2015 Red Hat, Inc. <http://www.redhat.com/>
--- glusterfs-3.12.4/geo-replication/syncdaemon/gsyncd.py.orig	2017-12-14 11:22:31.448031474 +0100
+++ glusterfs-3.12.4/geo-replication/syncdaemon/gsyncd.py	2017-12-26 11:05:36.587911481 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (c) 2011-2014 Red Hat, Inc. <http://www.redhat.com>
 # This file is part of GlusterFS.
--- glusterfs-3.12.4/geo-replication/syncdaemon/changelogagent.py.orig	2017-12-14 11:22:31.448031474 +0100
+++ glusterfs-3.12.4/geo-replication/syncdaemon/changelogagent.py	2017-12-26 11:05:48.794578005 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (c) 2011-2014 Red Hat, Inc. <http://www.redhat.com>
 # This file is part of GlusterFS.
--- glusterfs-3.12.4/geo-replication/syncdaemon/gsyncdstatus.py.orig	2017-12-14 11:22:31.448031474 +0100
+++ glusterfs-3.12.4/geo-replication/syncdaemon/gsyncdstatus.py	2017-12-26 11:06:04.354577832 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 #
 # Copyright (c) 2011-2014 Red Hat, Inc. <http://www.redhat.com>
 # This file is part of GlusterFS.
--- glusterfs-3.12.4/geo-replication/src/peer_georep-sshkey.py.in.orig	2017-12-14 11:22:31.446031470 +0100
+++ glusterfs-3.12.4/geo-replication/src/peer_georep-sshkey.py.in	2017-12-26 11:06:26.764577588 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/geo-replication/src/peer_mountbroker.py.in.orig	2017-12-14 11:22:31.447031472 +0100
+++ glusterfs-3.12.4/geo-replication/src/peer_mountbroker.py.in	2017-12-26 11:06:40.887910749 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 import os
 from errno import EEXIST, ENOENT
 
--- glusterfs-3.12.4/events/eventskeygen.py.orig	2017-12-14 11:22:31.433031442 +0100
+++ glusterfs-3.12.4/events/eventskeygen.py	2017-12-26 11:07:24.924576912 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/events/tools/eventsdash.py.orig	2017-12-14 11:22:31.434031444 +0100
+++ glusterfs-3.12.4/events/tools/eventsdash.py	2017-12-26 11:07:47.844576651 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/events/src/glustereventsd.py.orig	2017-12-14 11:22:31.433031442 +0100
+++ glusterfs-3.12.4/events/src/glustereventsd.py	2017-12-26 11:08:06.877909766 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/events/src/peer_eventsapi.py.orig	2017-12-14 11:22:31.434031444 +0100
+++ glusterfs-3.12.4/events/src/peer_eventsapi.py	2017-12-26 11:08:18.717909931 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/events/src/gf_event.py.orig	2017-12-14 11:22:31.433031442 +0100
+++ glusterfs-3.12.4/events/src/gf_event.py	2017-12-26 11:08:29.301242844 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 #  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
--- glusterfs-3.12.4/extras/hook-scripts/S40ufo-stop.py.orig	2017-12-14 11:22:31.440031457 +0100
+++ glusterfs-3.12.4/extras/hook-scripts/S40ufo-stop.py	2017-12-26 11:08:45.694575988 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 import os
 from optparse import OptionParser
